rose/server/player_test.py

```diff
@@ -1,15 +1,88 @@
-from . import player
+import pytest
+from rose.common import actions, config
+from rose.server.player import Player
 
 
-def test_in_lane():
-    p = player.Player("A", car=0, lane=0)
-    for x in (0, 1, 2):
-        p.x = x
-        assert p.in_lane()
+# Sample Test Setup
+@pytest.fixture
+def player():
+    return Player("John", "car1", 1)
 
 
-def test_not_in_lane():
-    p = player.Player("A", car=0, lane=1)
-    for x in (0, 1, 2):
-        p.x = x
-        assert not p.in_lane()
+def test_player_initialization(player):
+    assert player.name == "John"
+    assert player.car == "car1"
+    assert player.lane == 1
+    assert player.x is not None  # This checks that x is initialized
+    assert player.y == config.matrix_height // 3 * 2
+    assert player.action == actions.NONE
+    assert player.response_time == 1.0
```

Comment:
Strange that we have a value for new player that never had any response. Good that you added the test to reveal this issue.

Reply:
:-), changed the init method to start with None, and updated the test also